src/backend/cuda/hooks/yaksuri_cuda_init_hooks.c

```diff
@@ -113,6 +113,29 @@ int yaksuri_cuda_init_hook(yaksur_gpudriver_hooks_s ** hooks)
     cerr = cudaGetDeviceCount(&yaksuri_cudai_global.ndevices);
     YAKSURI_CUDAI_CUDA_ERR_CHKANDJUMP(cerr, rc, fn_fail);
 
+    if (getenv("CUDA_VISIBLE_DEVICES") == NULL) {
```

Comment:
The logic looks good to me to quickly fix the current runtime error. There seems no way to programmatically check the `CUDA_VISIBLE_DEVICES` property via CUDA API.
My concern is that this mechanism by default silently disables CUDA support. It would be user-friendly if it shows some hints (as warning if it is not an error) to enable it because I believe most users who use CUDA-enabled MPICH expect that this CUDA support is enabled.

Reply:
Yeah, I was considering that, but it would throw a warning even in the correct case that I mentioned above.  I don't like incorrect warnings, but I guess I also don't like silently disabling something without telling the user.  Maybe I can throw a warning and ask the user to set `CUDA_VISIBLE_DEVICES` to disable the warning.  Let me add that fix.

Reply:
@shintaro-iwasaki  Is this version better?

Reply:
Is setting CUDA_VISIBLE_DEVICES silence this warning? I believe users need to set CUDA_VISIBLE_DEVICES properly to enable CUDA support.

Reply:
It disables this warning too.  For example, one can set `CUDA_VISIBLE_DEVICES=-1` to disable this warning, but still not get CUDA support.  So I think the description is correct.

Reply:
I got it. Thanks for clarification.